Odata-docs/webapi-8/fundamentals/client-driven-paging.md

```diff
@@ -0,0 +1,173 @@
+```
+
+## Benefits of this approach
+
+This client-driven paging approach has a number of benefits:
+- It's relatively easy to implement
+- It allows the client to fetch pages independently and to jump to an arbitrary page. For example, the client can fetch page 5 without having fetched page 4 previously.
+- If the client knows the total size of the collection (e.g. using the `$count` query option), the client can compute the total number of pages as well as generate links for each page in advance.
+
+## Drawbacks
+
+Paging based on `$top` and `$skip` has a number of drawbacks that may make unsuitable in some scenarios.
+
+This approach is sensitive to changes in that data source which may lead the page contents to be unstable. To illustrate this, let's assume we have a data source with the following 5 items and we want to fetch 2 items per page:
+```json
+[
+    { "Id": 1, "Name": "Product 1" },
+    { "Id": 2, "Name": "Product 2" },
+    { "Id": 3, "Name": "Product 3" },
+    { "Id": 4, "Name": "Product 4" },
+    { "Id": 5, "Name": "Product 5" }
+]
+```
+The client can fetch the first page with 
+
+```http
+GET /Products?$top=2
+```
+This returns the following result:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        { "Id": 1, "Name": "Product 1" },
+        { "Id": 2, "Name": "Product 2" }
+    ]
+}
+```
+
+Let's say that product 2 is removed from the data source before the client fetches the second page. The data source looks like:
+```json
+[
+    { "Id": 1, "Name": "Product 1" },
+    { "Id": 3, "Name": "Product 3" },
+    { "Id": 4, "Name": "Product 4" },
+    { "Id": 5, "Name": "Product 5" }
+]
+```
+
+The client can fetch the second page with
+
+```http
+GET /Products?$top=2&$skip=2
+```
+
+This returns the following result:
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        { "Id": 4, "Name": "Product 4" },
+        { "Id": 5, "Name": "Product 5" }
+    ]
+}
+```
+
+Notice the client did not see product 3 because it was removed before it fetched page 2. In a similar way, if new products were added to the data source before product 2, then the client would see product 2 again in page 2 or later pages.
+
+If the underlying data source collection is not ordered, you may also need to manually order the results to have stable pages, e.g. `GET /Products?$top=2&$skip=2&$orderby=Id`.
+
+Using `$skip` can also lead to performance issues in large datasets in some data source implementation. In many SQL-based database systems `$skip` is implemented using `OFFSET`. A request like `/Products?$skip=50000&$top=100` could be translated in the following SQL query:
+
+```sql
+SELECT * FROM products
+LIMIT 100, OFFSET 50000;
+```
+This query will return 100 rows, from row 50,001 to row 50,100. However it will scan through all the first 50,100 rows in the table, then drop the first 50,000. The larger the offset, the longer the query will take.
```

Comment:
This explains why `skip` isn't suitable but I was looking (here and in server-side doc) for the explanation of how `skipToken` works internally so that it doesn't suffer the same issues.

Reply:
See: https://github.com/MicrosoftDocs/OData-docs/pull/281#discussion_r1087443168